pwr.md

```diff
@@ -0,0 +1,147 @@
+## Refreshing cpu stats
+CPU stats can become out of date, such as `curr_freq` or `sst_bf_configured`. These can be refreshed in with `refresh_stats()`.
+It is *advised* that CPU and core objects be updated with `refresh_stats()` before requesting object data.
+
+core.refresh_stats() will update:
+* `curr_freq`             
+* `min_freq`                
+* `max_freq`                
+* `epp`
+                
+cpu.refresh_stats() will update:
+* `uncore_freq`
+* `uncore_max_freq` 
+* `uncore_min_freq`
+* `sst_bf_configured`
+
+```python
+for c in cores:
+    c.refresh_stats()  # Refresh core stats, which refreshes the sst_bf_configured value
+    if not c.sst_bf_configured:  # Check are cores configured for SST-BF
+        c.commit("sst_bf_base")  # Set cores to SST-BF configuration
+```
+
+## Object Referencing
+Both CPU & core objects reference each other, the list of core objects can be accessed through the CPU and the CPU object of a core can be accessed through a core.
+
+```python
+for core in cores:  # Loop through cores in object list 
+    if core.cpu.sst_bf_enabled:  # Access CPU flag from core object
+        if core.high_priority:  # Check if core has high priority base frequency 
+            core.min_freq = core.base_freq  # Modify minimum desired frequency
+            core.max_freq = core.sst_bf_base_freq  # Modify maximum desired frequency
+            core.commit()  # Commit changes to hardware
+```
+or
+```python
+for cpu in cpus:
+    if cpu.core_list[0].epp == "performance":  # Access core performance profile through core list of CPU object
+        cpu.uncore_max_freq = cpu.uncore_hw_max  # set desired max uncore frequency from hardware limit
+        cpu.uncore_min_freq = cpu.uncore_hw_min  # set desired min uncore frequency from hardware limit
+        cpu.commit() #Commit changes to hardware
+```
+
+## Power Consumption
+The power consumption of the CPU can be read from the `power_consumption` attribute, this value can be compared against the `tdp` value to check is the current power draw close to the limit, indicated by the tdp value. The power consumption is reported as average since last time it was calculated. The first time this value is read, it can be 0. The next time it is read, it will show average power consumption (in Watts) since last read. Time period between reads must not exceed 60 seconds, otherwise the value will be reset to 0, hence, why the initial read may be 0, depending on time between initilization and the first read. 
+
+## Concept Overview
+### CPU
+A CPU may have multiple physical cores, which are represented by the core objects. These cores can optionally have multiple threads, which means there would be multiple logical cores on a single physical core, these are called thread siblings. Each logical core will also have its own core object. Uncore frequency is the frequency at which everything on the CPU, except the cores, runs at. This can be scaled within the limits of min and max. 
+### Core
```

Comment:
I don't see us mentioning possible EPP values anywhere? It was there in the first version, but got lost somewhere along the way.

Reply:
I think the best place to put it would be alongside the list of profiles.